src/main/java/mat/client/shared/CQLLibraryResultTable.java

```diff
@@ -62,41 +64,75 @@ public Boolean getValue(CQLLibraryDataSetObject object) {
 			public SafeHtml getValue(CQLLibraryDataSetObject object) {
 				return getCQLLibraryNameColumnToolTip(object);
 			}
-		};
-//		Double Click event on Grid row to navigate to Composer Tab.
-		if(MatContext.get().getMatOnFHIR().getFlagOn()) {
-			table.addDomHandler(new DoubleClickHandler() {
-				@Override
-				public void onDoubleClick(DoubleClickEvent event) {
-					for (CQLLibraryDataSetObject object : selectionModel.getSelectedSet()) {
-						SelectionEvent.fire(fireEvent, object);
-					}
+			/*
+				Single Click to select row and enable checkbox
+				Double Click to navigate to CQL Composer tab
+			*/
+			@Override
+			public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
+				if (lastClick < System.currentTimeMillis() - 500)
+				{
+					object.setSelected(!object.isSelected());
+					selectionModel.setSelected(object, object.isSelected());
+				}else{
+					SelectionEvent.fire(fireEvent, object);
 				}
-			}, DoubleClickEvent.getType());
-		}
+				lastClick = System.currentTimeMillis();
+			}
+		};
 		table.addColumn(cqlLibraryName,
 				SafeHtmlUtils.fromSafeConstant("<span title='CQL Library Name'>" + "CQL Library Name" + "</span>"));
 
 		// Version Column
 		Column<CQLLibraryDataSetObject, SafeHtml> version = new Column<CQLLibraryDataSetObject, SafeHtml>(
-				new MatSafeHTMLCell()) {
+				new ClickableSafeHtmlCell()) {
 			@Override
 			public SafeHtml getValue(CQLLibraryDataSetObject object) {
 				return CellTableUtility.getColumnToolTip(object.getVersion());
 			}
+			/*
+				Single Click to select row and enable checkbox
+				Double Click to navigate to CQL Composer tab
+			*/
+			@Override
+			public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
+				if (lastClick < System.currentTimeMillis() - 500)
```

Comment:
The state of lastClick is shared by different instances of event handlers for different columns. can it cause any UI glitches?

Reply:
you meant to say that, I have to restrict the state of lastClick for every object?